File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;


class Solution {
    public int solution(int[] nums) {
        int n = nums.length/2;
        int answer = 0;
        HashSet<Integer> set = new HashSet<Integer>();
        
        for(int i = 0; i<nums.length; i++) {
            set.add(nums[i]);
        }
        
        if (set.size() < n) {
            answer = set.size();
        }
        else {
            answer = n;
        }
        
        return answer;
    }
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int visit[200001] = {0,};
    
    for(int i = 0; i<nums.size(); i++) {
        if (visit[nums[i]] == 0 && answer < nums.size()/2) {
            visit[nums[i]] = 1;
            answer++;
        }
    }
    
    return answer;
}